### ISSUE TITLE
Default presentation not show with https

### ISSUE THREAD

**Issue by lgmatabh** [bigbluebutton/bigbluebutton#6139]
hello,

i configured with sucess the BBB and default presentation not has show. The certificate is validated in browser but in javascript occurs error:

Oct  8 10:01:25 webconferenciads01 systemd_start.sh[1243]: #033[31merror#033[39m: Error parsing image size. RequestError: unable to verify the first certificate. uri=https://webconferenciads01.pbh.gov.br/bigbluebutton/presentation/183f0bf3a0982a127bdb8161e0c44eb696b3e75c-1539003676413/183f0bf3a0982a127bdb8161e0c44eb696b3e75c-1539003676413/d2d9a672040fbde2a47a10bf6c37b6a4b5ae187f-1539003676806/svg/5
Oct  8 10:01:25 webconferenciads01 systemd_start.sh[1243]: #033[31merror#033[39m: Error parsing image size. RequestError: unable to verify the first certificate. uri=https://webconferenciads01.pbh.gov.br/bigbluebutton/presentation/183f0bf3a0982a127bdb8161e0c44eb696b3e75c-1539003676413/183f0bf3a0982a127bdb8161e0c44eb696b3e75c-1539003676413/d2d9a672040fbde2a47a10bf6c37b6a4b5ae187f-1539003676806/svg/2
Oct  8 10:01:25 webconferenciads01 systemd_start.sh[1243]: #033[31merror#033[39m: Error parsing image size. RequestError: unable to verify the first certificate. uri=https://webconferenciads01.pbh.gov.br/bigbluebutton/presentation/183f0bf3a0982a127bdb8161e0c44eb696b3e75c-1539003676413/183f0bf3a0982a127bdb8161e0c44eb696b3e75c-1539003676413/d2d9a672040fbde2a47a10bf6c37b6a4b5ae187f-1539003676806/svg/4
Oct  8 10:01:25 webconferenciads01 systemd_start.sh[1243]: #033[31merror#033[39m: Error parsing image size. RequestError: unable to verify the first certificate. uri=https://webconferenciads01.pbh.gov.br/bigbluebutton/presentation/183f0bf3a0982a127bdb8161e0c44eb696b3e75c-1539003676413/183f0bf3a0982a127bdb8161e0c44eb696b3e75c-1539003676413/d2d9a672040fbde2a47a10bf6c37b6a4b5ae187f-1539003676806/svg/6
Oct  8 10:01:25 webconferenciads01 systemd_start.sh[1243]: #033[31merror#033[39m: Error parsing image size. RequestError: unable to verify the first certificate. uri=https://webconferenciads01.pbh.gov.br/bigbluebutton/presentation/183f0bf3a0982a127bdb8161e0c44eb696b3e75c-1539003676413/183f0bf3a0982a127bdb8161e0c44eb696b3e75c-1539003676413/d2d9a672040fbde2a47a10bf6c37b6a4b5ae187f-1539003676806/svg/1

![https](https://user-images.githubusercontent.com/39322388/46612880-22cec600-cae8-11e8-94d3-6b91b110b546.png)
![page_bbb](https://user-images.githubusercontent.com/39322388/46612990-5c073600-cae8-11e8-943f-f7740d403078.png)

**Comment by ffdixon** [comment]
Can you try the following on the command line (from an SSH prompt on your BigBlueButton server).

  wget https://<your_server>/default.pdf

and try opening the above URL in your browser.

Do either of those two accesses to default.pdf yield an error?

**Comment by lgmatabh** [comment]
Hi ffdixon,

wget successfully, despite not having the certifying authority for wget.

gonzaga @ pc158122: ~ $ wget https://webconferenciads01.pbh.gov.br/default.pdf
--2018-10-08 11: 08: 12-- https://webconferenciads01.pbh.gov.br/default.pdf
Resolving netuno.pbh.gov.br (netuno.pbh.gov.br) ... 10.0.24.179
Connecting to the.wikipedia.org/wiki/index.php?
WARNING: The certificate of "webconferenciads01.pbh.gov.br" is not trustworthy.
WARNING: The certificate of "webconferenciads01.pbh.gov.br" does not have a known issuer.
The Proxy request was sent, awaiting response ... 200 OK
Size: 369690 (361K) [application / pdf]
Saving in: "default.pdf"

default.pdf                             100%[=============================================================================>] 361,03K  --.-KB/s    in 0,04s   

2018-10-08 11:08:13 (8,30 MB/s) - “default.pdf” salvo [369690/369690]

**Comment by ffdixon** [comment]
> WARNING: The certificate of "webconferenciads01.pbh.gov.br" is not trustworthy.

While wget may download it, the BigBlueButton server (specifically bbb-web running in tomcat7) is trying to access this URL and is failing due to the same error.

> Connecting to the.wikipedia.org/wiki/index.php?

That seems to be a strange connection .. are you sure the file is a PDF?

Here's what a valid access should look like:

~~~
# wget https://demo.bigbluebutton.org/default.pdf
--2018-10-08 10:16:26--  https://demo.bigbluebutton.org/default.pdf
Resolving demo.bigbluebutton.org (demo.bigbluebutton.org)... 2607:5300:60:1e50::, 198.27.69.80
Connecting to demo.bigbluebutton.org (demo.bigbluebutton.org)|2607:5300:60:1e50::|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 369690 (361K) [application/pdf]
Saving to: ‘default.pdf’

default.pdf                                   100%[===============================================================================================>] 361.03K  --.-KB/s    in 0.001s

2018-10-08 10:16:26 (379 MB/s) - ‘default.pdf’ saved [369690/369690]
~~~

**Comment by lgmatabh** [comment]
With the option  --no-proxy in wget in internal network 

gonzaga @ pc158122: ~ $ wget https://webconferenciads01.pbh.gov.br/default.pdf --no-proxy
--2018-10-08 11: 24: 26-- https://webconferenciads01.pbh.gov.br/default.pdf
Resolving webconferenciads01.pbh.gov.br (webconferenciads01.pbh.gov.br) ... 10.0.26.45
Connecting to webconferenciads01.pbh.gov.br (webconferenciads01.pbh.gov.br) | 10.0.26.45 |: 443 ... connected.
WARNING: The certificate of "webconferenciads01.pbh.gov.br" is not trustworthy.
WARNING: The certificate of "webconferenciads01.pbh.gov.br" does not have a known issuer.
HTTP request sent, waiting for response ... 200 OK
Size: 369690 (361K) [application / pdf]
Saving in: "default.pdf.2"

default.pdf.2 100% [================================================== =======================================] 04s

2018-10-08 11:24:26 (8.44 MB / s) - "default.pdf.2" saved [369690/369690]

**Comment by ffdixon** [comment]
If the certificate for webconferenciads01.pbh.gov.br is a valid certificate (not a self-signed certificate), you'll need to add the root certificate to the Java Certificate Authority store.  If you google for `java certificate strore` you'll get lots of hits on how to do this.

**Comment by lgmatabh** [comment]
Already configured this.

The message "Error parsing image size. RequestError: unable to verify the first certificate."  still occurs

Configuration in server.xml:

 <Connector port="8080" protocol="HTTP/1.1"
               connectionTimeout="20000"
               URIEncoding="UTF-8"
               redirectPort="8443" secure="true" />
        <Valve className="org.apache.catalina.valves.RemoteIpValve" remoteIpHeader="x-forwarded-for" proxiesHeader="x-forwarded-by" protocolHeader="x-forwarded-proto" />

        <Connector port="8443" protocol="HTTP/1.1" SSLEnabled="true"
               maxThreads="1000" scheme="https" secure="true"
               maxHttpHeaderSize="8192"
               clientAuth="false" sslProtocol="TLS" connectionTimeout="60000"
               keystoreFile="/usr/local/certificado/pbh.keystore" keystorePass="xxxxxx"/>

I also configured cacerts in Java, but bbb not showing the default apresentation.

Is there a way to configure the parameter to accept the certificate in node.js, for example?

ex: https://stackoverflow.com/questions/36194326/request-error-unable-to-verify-the-first-certificate

**Comment by ffdixon** [comment]
> Is there a way to configure the parameter to accept the certificate in node.js, for example?

I don't know how to do this ... also, your site isn't reachable externally so I wasn't able to check out the setup using sslabs.  Also, I did a quick google for "default.pdf bigbluebutton certificate" and found a related thread

  https://groups.google.com/forum/#!topic/bigbluebutton-dev/KOm5tVEzOAQ

**Comment by lgmatabh** [comment]
> I don't know how to do this ... also, your site isn't reachable externally so I wasn't able to check out the setup using sslabs. Also, I did a quick google for "default.pdf bigbluebutton certificate" and found a related thread

This BBB is in intranet with domain pbh.gov.br in DNS internal (webconferenciads.pbh.gov.br -> 10.0.26.45) for use de validated certificate in corporative network ,not in Internet. This is the reason you did not reach it!

It seems that javascript can not validate the certificate as a BBB client, where could I put the parameter ("rejectUnauthorized": false) in the BBB client for testing? The correct option is configure the chain of certificate authority in Node.

https://stackoverflow.com/questions/36194326/request-error-unable-to-verify-the-first-certificate

**Comment by lgmatabh** [comment]
More information:

One of these generates the error message:

root@webconferenciads01:/usr# **grep -r "unable to verify the first certi" ***
Binary file bin/node matches
lib/x86_64-linux-gnu/perl5/5.22/Net/SSLeay.pod: #     21 - X509_V_ERR_UNABLE_TO_VERIFY_LEAF_SIGNATURE: unable to verify the first certificate
Binary file lib/x86_64-linux-gnu/libcrypto.a matches
Binary file share/node-v8.9.4-linux-x64/bin/node matches
src/node-v8.9.4-linux-x64/lib/node_modules/npm/TROUBLESHOOTING.md:npm ERR! unable to verify the first certificate
Binary file src/node-v8.9.4-linux-x64/bin/node matches

**Comment by capilkey** [comment]
Turning off the security check would compromise the application. I think the correct way to solve the problem would be to add the certificate to node using the NODE_EXTRA_CA_CERTS environment variable. See this SO post, https://stackoverflow.com/a/47160447.

**Comment by lgmatabh** [comment]
Hi Capilkey,

I exported CA of  Comodo on browser, i, also, exported the environment variable 'export  NODE_EXTRA_CA_CERTS=/usr/local/certificado/COMODORSAOrganizationValidationSecureServerCA.crt"  and the error continue.

Oct  9 15:40:10 webconferenciads01 systemd_start.sh[1468]: #033[31merror#033[39m: Error parsing image size. RequestError: unable to verify the first certificate. uri=https://webconferenciads01.pbh.gov.br/bigbluebutton/presentation/183f0bf3a0982a127bdb8161e0c44eb696b3e75c-1539110400792/183f0bf3a0982a127bdb8161e0c44eb696b3e75c-1539110400792/d2d9a672040fbde2a47a10bf6c37b6a4b5ae187f-1539110401568/svg/5

**Comment by capilkey** [comment]
Did you restart BBB after adding the environment variable? Node.js probably only looks for the environment variable on start up.

**Comment by capilkey** [comment]
I can't read Portuguese, but that wget output doesn't look okay. Looks like there are two warnings/errors about the certificate. Seems like to fix wget you'd need to add the certificate under "/usr/lib/ssl/certs" because that's where openssl (which wget should be using by default) will look for CAs. It will probably just fix the wget output though and not the core node.js problem. 

About the node evironment variable, the documentation suggests that the certificate must be in PEM form. See, https://nodejs.org/api/cli.html#cli_node_extra_ca_certs_file.

**Comment by lgmatabh** [comment]
Hello,

my scenario is:

A) BBB with IPv4 - 10.0.26.45 - and clients in the same local corporate network;

B) There is an edge firewall that has no valid certificate;

C) BBB working properly with https using players plugin;

D) With html5 the error "systemd_start.sh [1213] occurs: # 033 [31th error # 033 [39m: Error while parsing image size. RequestError: Unable to verify the first certificate. Uri = https: // ..." with the request of the http.request of the Javascript arriving at the default gateway of the network that is the edge firewall with certificate invalid.


After investigation, many tests and a new installation successfully using https and using the client player, I continue with the certificate error using html5.

I configured environment variables http_proxy, https_proxy and non_proxy without success in linux.

The solution seems to be described at this link: https://github.com/npm/cli/commit/0e576f0aa6ea02653d948c10f29102a2d4a31944

The version in the Node embedded in the bbb does not seem to be with the above implementation and no_proxy does not work.

root @ webconferenciads01: / usr / local # grep -ri no_proxy *
bigbluebutton / bbb-webrtc-sfu / node_modules / utf-8-validate / build / config.gypi: "no_proxy": "",
bigbluebutton / bbb-webrtc-sfu / node_modules / websocket-stream / node_modules / ws / build / config.gypi: "no_proxy": "",
bigbluebutton / bbb-webrtc-sfu / node_modules / sip.js / node_modules / ws / build / config.gypi: "no_proxy": "",

I need to put my corporate network 10.0.0.0/8 as npm's noproxy exception.

Att"",

**Comment by capilkey** [comment]
I don't think "no_proxy" in npm would have any relevance here. It seems like that setting is for downloading node packages through a proxy, whereas the problem you're hitting is when the running application tries to download the presentation SVGs. 

I still think the problem is going to be something about misconfigured or malformed certificates.

**Comment by lgmatabh** [comment]
Hello Capilkey,

> I don't think "no_proxy" in npm would have any relevance here. It seems like that setting is for downloading node packages through a proxy, whereas the problem you're hitting is when the running application tries to download the presentation SVGs.
> 
> I still think the problem is going to be something about misconfigured or malformed certificates.

You are correct, it is a certificate configuration problem in nginx.

**Problem solved!**

When tracing the error, I came across the "https://www.npmjs.com/package/ssl-root-cas#unable-to-verify-the-first-certificate" link, where it explicitly says the following: "did do not include the intermediate certificates in the certificate it sent you." for this error.

I created the .pem file with the public key of the site + intermediate certificate.  I replaced the .crt file with the public key of the site certificate with the two public keys in a .pem file.

The interesting thing is that and the documentation 

A) For the use of the BBB with plugin in the browser, just the public key of the site certificate in nginx for the perfect operation.

B) **For html5 and Node,** public key information for the site and intermediate CA  is required in a single .pem file.

#ssl_certificate /etc/nginx/ssl/nw-pbh.gov.br.crt; 
ssl_certificate /etc/nginx/ssl/com_intermediate_pbh.gov.br.pem;

**Comment by capilkey** [comment]
I'm glad to hear that you got it to work. I'm closing this issue now.

**Comment by lnxbil** [comment]
I trapped into the same problem with a self-signed CA infrastructure, but my problem was a bit different to solve:

I use debian-packages to distribute my root and intermediate certificates that plugs directly into the `ca-certificates` chain, so that `bbb` works mostly, the only problem is the `nodejs` metero stuff, that does not use the os-based `ca-certificates`, I needed to edit the file `/usr/share/meteor/bundle/systemd_start.sh` and add in the second last line the full path to my root certificate: `export NODE_EXTRA_CA_CERTS=/usr/share/ca-certificates/<full-path>`. After that I could use my own certificates without a problem and the presentation will work like they should.

For greenlight, I just created a new overlay like this:

```Dockerfile
FROM bigbluebutton/greenlight:v2

COPY files/*crt /usr/local/share/ca-certificates/
RUN update-ca-certificates
```

and everything is up an running with self-signed certificates.

**Comment by MeinhartEsrohr** [comment]
Hello Inxbil, 

I've been struggling with the certificate for several days. I have a question to you: 
did you specify your root certificate (export command : /usr/share/meteor/bundle/systemd_start.sh) in. * crt or in * .pem format?

If I look under  / usr / share / ca-certificates,  there are no certificates except one Mozilla folder.
I just simply specified the path to my root certificate:
 export NODE_EXTRA_CA_CERTS=/home/ms/root.crt
My basic problem is that I also have the problem with the microphone and the presentation is not displayed.

**Comment by lnxbil** [comment]
Hi Mainhart,

> I've been struggling with the certificate for several days. I have a question to you:
> did you specify your root certificate (export command : /usr/share/meteor/bundle/systemd_start.sh) in. * crt or in * .pem format?

PEM and it is the root certificate. On the webserver side, I also have the webserver certificate and the immediate one, so that the whole chain is present and readable by meteor.

> If I look under / usr / share / ca-certificates, there are no certificates except one Mozilla folder.
> I just simply specified the path to my root certificate:
> export NODE_EXTRA_CA_CERTS=/home/ms/root.crt

I have all my root certifcates (3 root, 2 intermediate) packaged and installed in `/usr/share/ca-certificates/<mycomany>` so that everything on the system, that uses the system-wide ca-stuff that is automatically merged into the system keychain in `/etc/ssl/certs/ca-certificates.crt`.

> My basic problem is that I also have the problem with the microphone and the presentation is not displayed.

The microphone is normally a problem with encryption of a webserver itself. All browsers I know do not allow unencrypted microphone and webcam transmisson. Have you installed certificates for nginx? If not, you need to [do so](https://docs.bigbluebutton.org/2.2/install.html#configure-ssl-on-your-bigbluebutton-server).

**Comment by MeinhartEsrohr** [comment]
> Hi Mainhart,
> 
> > I've been struggling with the certificate for several days. I have a question to you:
> > did you specify your root certificate (export command : /usr/share/meteor/bundle/systemd_start.sh) in. * crt or in * .pem format?
> 
> PEM and it is the root certificate. On the webserver side, I also have the webserver certificate and the immediate one, so that the whole chain is present and readable by meteor.
> 
> > If I look under / usr / share / ca-certificates, there are no certificates except one Mozilla folder.
> > I just simply specified the path to my root certificate:
> > export NODE_EXTRA_CA_CERTS=/home/ms/root.crt
> 
> I have all my root certifcates (3 root, 2 intermediate) packaged and installed in `/usr/share/ca-certificates/<mycomany>` so that everything on the system, that uses the system-wide ca-stuff that is automatically merged into the system keychain in `/etc/ssl/certs/ca-certificates.crt`.
> 
i unserstand, I read the following syntax in another thread, this has now worked for my presentation problem.

export NODE_EXTRA_CA_CERTS='/etc/ssl/certs/[myrootcert].pem'
bbb-conf --restart


> > My basic problem is that I also have the problem with the microphone and the presentation is not displayed.
> 
> The microphone is normally a problem with encryption of a webserver itself. All browsers I know do not allow unencrypted microphone and webcam transmisson. Have you installed certificates for nginx? If not, you need to [do so](https://docs.bigbluebutton.org/2.2/install.html#configure-ssl-on-your-bigbluebutton-server).

I have installed a certificate for nginx,i also tried to define my root CA 
/etc/nginx/sites-available/bigbluebutton
server {
     listen   80;
     listen [::]:80;
     server_name  192.168.2.1;
     listen 443 ssl;
     listen [::]:443 ssl;
     ssl_certificate /home/ms/Server01.crt;
     ssl_certificate_key /home/ms/Server01.key;
     ssl_client_certificate /home/ms/Root_CA.crt;
     ssl_session_cache shared:SSL:10m;
     ssl_session_timeout 10m;
     ssl_protocols TLSv1 TLSv1.1 TLSv1.2;
     ssl_ciphers "ECDH+AESGCM:DH+AESGCM:ECDH+AES256:DH+AES256:ECDH+AES128:DH+AES:ECDH+3DES:DH+3DES:RSA+AESGCM:RSA+AES:RSA+3DES:!aNULL:!MD5:!DSS:!AES256";
     ssl_prefer_server_ciphers on;
     ssl_dhparam /home/ms/dhp-4096.pem;


But i think my Microphone problems comes from here:
if i change the BBG Endpoint Link in . env File to HTTPS  and Check with: 
root@Server01:~/greenlight# docker run --rm --env-file .env bigbluebutton/greenlight:v2 bundle exec rake conf:check

i get this error:
Error connecting to BigBlueButton server - SSL_connect returned=1 errno=0 state=SSLv3 read server certificate B: certificate verify failed (unable to get local issuer certificate)

when i change it to http: it works

BIGBLUEBUTTON_ENDPOINT=http://192.168.2.1/bigbluebutton/
BIGBLUEBUTTON_SECRET=0zLEt8Hxfvy3EjTDt54ZpVVOHTKPegalMdY8IiJBoHH



I apologize my formatting, thank you so much for helping

**Comment by lnxbil** [comment]
> But i think my Microphone problems comes from here:
> if i change the BBG Endpoint Link in . env File to HTTPS and Check with:
> root@Server01:~/greenlight# docker run --rm --env-file .env bigbluebutton/greenlight:v2 bundle exec rake conf:check
> 
> i get this error:
> Error connecting to BigBlueButton server - SSL_connect returned=1 errno=0 state=SSLv3 read server certificate B: certificate verify failed (unable to get local issuer certificate)
> 
> when i change it to http: it works
> 
> BIGBLUEBUTTON_ENDPOINT=http://192.168.2.1/bigbluebutton/
> BIGBLUEBUTTON_SECRET=0zLEt8Hxfvy3EjTDt54ZpVVOHTKPegalMdY8IiJBoHH
> 
> I apologize my formatting, thank you so much for helping

For greenlight, you need to add the root certificate as well. I described it a few posts above

**Comment by MeinhartEsrohr** [comment]
> For greenlight, I just created a new overlay like this:
> 
> ```dockerfile
> FROM bigbluebutton/greenlight:v2
> 
> COPY files/*crt /usr/local/share/ca-certificates/
> RUN update-ca-certificates
> ```
> 
> and everything is up an running with self-signed certificates.

Really sorry, i dont understand what you mean with:

FROM bigbluebutton/greenlight:v2 
COPY files/*crt /usr/local/share/ca-certificates/
RUN update-ca-certificates


Could you  please explain that to me a little bit more ?

Thank you very much.

**Comment by lnxbil** [comment]
> > For greenlight, I just created a new overlay like this:
> > ```dockerfile
> > FROM bigbluebutton/greenlight:v2
> > 
> > COPY files/*crt /usr/local/share/ca-certificates/
> > RUN update-ca-certificates
> > ```
> > 
> > 
> > and everything is up an running with self-signed certificates.
> 
> Really sorry, i dont understand what you mean with:
> 
> FROM bigbluebutton/greenlight:v2
> COPY files/*crt /usr/local/share/ca-certificates/
> RUN update-ca-certificates
> 
> Could you please explain that to me a little bit more ?
> 
> Thank you very much.

You create a new docker image on base of the old one and just add your root ca.

Just copy&paste it into a new file called `Dockerfile`, create directory named `files`, copy your certificate files suffixed with `*.crt` in and build it with `docker build -t mygreenlight .`  Now you can use this new dockage image in your `docker-compose.yml`.

**Comment by MeinhartEsrohr** [comment]
Hey Inxbil,

thank you so much, but i think I can't do it , i dont have that knowledge, first time for me with docker, never worked with docker befor. have already wasted a lot of time with the CA issue.

Thank you

**Comment by lnxbil** [comment]
> thank you so much, but i think I can't do it , i dont have that knowledge, first time for me with docker, never worked with docker befor. have already wasted a lot of time with the CA issue.

Okay, understandable. What really works well is to just install on a root server with the official installer and use letsencrypt certificates. Works out of the box and really well.

**Comment by digitalsleuth** [comment]
FWIW, the intermediate certificates weren't what solved or caused the problem for me, as there were none created in my scenario. What I ended up having to do was:
in `/usr/share/meteor/bundle/systemd_start.sh` I added `export NODE_EXTRA_CA_CERTS=/usr/local/share/ca-certificates/myCA.pem`

Then made sure this, and the my.host.com.crt file, my dhp-4096.pem (for good measure) and the myCA.pem file were located both in `/usr/local/share/ca-certificates` and in `/etc/nginx/ssl` as the configs recommend.
Once that is done, `sudo update-ca-certificates` and you should have the certificate added.

During my fault-finding, I had also added the certs to /usr/share/ca-certificates just to be safe. I then issued the `sudo bbb-conf --restart` command and waited for it to reboot.

I ended up getting a ProbeError (404) after this, and that was because the presentation directories were incorrect. I had to:
- create a presentation directory at /var/bigbluebutton/ and `sudo chown bigbluebutton:bigbluebutton /var/bigbluebutton/presentation`
- edit `/usr/share/bbb-web/WEB-INF/classes/bigbluebutton.properties` and change:
    - presentationDir to `/var/bigbluebutton/presentation`
    - presentationBaseURL to `${bigbluebutton.web.serverURL}/bigbluebutton/presentation`
- edit `/usr/local/bigbluebutton/core/scripts/bigbluebutton.yml` and change/confirm raw_presentation_src was `/var/bigbluebutton/presentation`
and
- edit `/etc/bigbluebutton/nginx/presentation-slides.nginx` so that the line which says `alias /var/bigbluebutton/$meeting...` now said `alias /var/bigbluebutton/presentation` to match the `location ~^\/bigbluebutton\/presentation` line above it.

Now, this worked for my situation with all of the same problems you had. Took me two days and reading every related issue, but found the issues. Mostly resolved thanks to this thread and #8136 .

I hope this helps everyone, it definitely helped me.

**Comment by mmustak** [comment]
> > > For greenlight, I just created a new overlay like this:
> > > ```dockerfile
> > > FROM bigbluebutton/greenlight:v2
> > > 
> > > COPY files/*crt /usr/local/share/ca-certificates/
> > > RUN update-ca-certificates
> > > ```
> > > 
> > > 
> > > and everything is up an running with self-signed certificates.
> > 
> > 
> > Really sorry, i dont understand what you mean with:
> > FROM bigbluebutton/greenlight:v2
> > COPY files/*crt /usr/local/share/ca-certificates/
> > RUN update-ca-certificates
> > Could you please explain that to me a little bit more ?
> > Thank you very much.
> 
> You create a new docker image on base of the old one and just add your root ca.
> 
> Just copy&paste it into a new file called `Dockerfile`, create directory named `files`, copy your certificate files suffixed with `*.crt` in and build it with `docker build -t mygreenlight .` Now you can use this new dockage image in your `docker-compose.yml`.

Hi @lnxbil 
i have try your solution about greenlight, I think because this issue is about greenlight, how if we move to https://github.com/bigbluebutton/greenlight/issues/1305 . In there i have been explain my problem, but the others surrender about my problem.

**Comment by BehradJafari** [comment]
thanks for all the answers. but there is a simple way to solve this problem.
try to re-run your command for installing SSL and setting your domain.
e.g : `wget -qO- https://ubuntu.bigbluebutton.org/bbb-install.sh | bash -s -- -v xenial-220 -s bbb.example.com -e info@example.com`

**Comment by v2e** [comment]
# As of September 30, 2021
this may be related to the expiration of the [DST Root CA X3 certificate](https://letsencrypt.org/docs/dst-root-ca-x3-expiration-september-2021).

### To check if this is the case
run
`journalctl -f -u bbb-html5`
and restart BigBlueButton:
`bbb-config --restart`
If you see the lines like

> systemd_start.sh[30265]: error: No file found. Error: certificate has expired
> systemd_start.sh[30265]: error: Error parsing image size. Error: certificate has expired.

then it may be the case.
### To fix the problem

1. Make sure the ISRG Root X1 certificate is present on your system:
    `grep ISRG_Root_X1 /etc/ca-certificates.conf`
   If it is not there, you should upgrade the ca-certificates package and the OpenSSL to the latest version possible.
2. Comment out the expired DST Root CA X3 certificate in `/etc/ca-certificates.conf` with an exclamation mark like this:
    `!mozilla/DST_Root_CA_X3.crt`
3. Update the ca-certificates:
    `sudo update-ca-certificates`
4. Make the Node.js use the system-wide CA certificates. For this purpose at the end of the file `/usr/share/meteor/bundle/systemd_start.sh` add the parameter `--use-openssl-ca` like this:
    `PORT=3000 /usr/share/$NODE_VERSION/bin/node --use-openssl-ca --max-old-space-size=4096 --max_semi_space_size=128  main.js`

After that restart the BigBlueButton again, start some session and watch the logs. The lines about the expired certificate should not appear. At least, it worked in my case. Hope that helps somebody.

**Comment by MudrekaBohra** [comment]
> # As of September 30, 2021
> this may be related to the expiration of the [DST Root CA X3 certificate](https://letsencrypt.org/docs/dst-root-ca-x3-expiration-september-2021).
> 
> ### To check if this is the case
> run `journalctl -f -u bbb-html5` and restart BigBlueButton: `bbb-config --restart` If you see the lines like
> 
> > systemd_start.sh[30265]: error: No file found. Error: certificate has expired
> > systemd_start.sh[30265]: error: Error parsing image size. Error: certificate has expired.
> 
> then it may be the case.
> 
> ### To fix the problem
> 1. Make sure the ISRG Root X1 certificate is present on your system:
>    `grep ISRG_Root_X1 /etc/ca-certificates.conf`
>    If it is not there, you should upgrade the ca-certificates package and the OpenSSL to the latest version possible.
> 2. Comment out the expired DST Root CA X3 certificate in `/etc/ca-certificates.conf` with an exclamation mark like this:
>    `!mozilla/DST_Root_CA_X3.crt`
> 3. Update the ca-certificates:
>    `sudo update-ca-certificates`
> 4. Make the Node.js use the system-wide CA certificates. For this purpose at the end of the file `/usr/share/meteor/bundle/systemd_start.sh` add the parameter `--use-openssl-ca` like this:
>    `PORT=3000 /usr/share/$NODE_VERSION/bin/node --use-openssl-ca --max-old-space-size=4096 --max_semi_space_size=128  main.js`
> 
> After that restart the BigBlueButton again, start some session and watch the logs. The lines about the expired certificate should not appear. At least, it worked in my case. Hope that helps somebody.

This worked like a charm on 2.2 installations
Thank you!

**Comment by fattahi** [comment]
> # As of September 30, 2021
> this may be related to the expiration of the [DST Root CA X3 certificate](https://letsencrypt.org/docs/dst-root-ca-x3-expiration-september-2021).
> 
> ### To check if this is the case
> run `journalctl -f -u bbb-html5` and restart BigBlueButton: `bbb-config --restart` If you see the lines like
> 
> > systemd_start.sh[30265]: error: No file found. Error: certificate has expired
> > systemd_start.sh[30265]: error: Error parsing image size. Error: certificate has expired.
> 
> then it may be the case.
> 
> ### To fix the problem
> 1. Make sure the ISRG Root X1 certificate is present on your system:
>    `grep ISRG_Root_X1 /etc/ca-certificates.conf`
>    If it is not there, you should upgrade the ca-certificates package and the OpenSSL to the latest version possible.
> 2. Comment out the expired DST Root CA X3 certificate in `/etc/ca-certificates.conf` with an exclamation mark like this:
>    `!mozilla/DST_Root_CA_X3.crt`
> 3. Update the ca-certificates:
>    `sudo update-ca-certificates`
> 4. Make the Node.js use the system-wide CA certificates. For this purpose at the end of the file `/usr/share/meteor/bundle/systemd_start.sh` add the parameter `--use-openssl-ca` like this:
>    `PORT=3000 /usr/share/$NODE_VERSION/bin/node --use-openssl-ca --max-old-space-size=4096 --max_semi_space_size=128  main.js`
> 
> After that restart the BigBlueButton again, start some session and watch the logs. The lines about the expired certificate should not appear. At least, it worked in my case. Hope that helps somebody.
Thank you for your help.
After apply changes it's enough to restart **html5** service. restarting **bbb** make latency in live services.
`systemctl restart bbb-html5.service`

**Comment by lookas1** [comment]
The above fixes didn't help in my case (I'm using Certum SSL certificates), however I found confirmed bug in nodejs 8.

So to easily test this issue one can run _node_ and then in nodejs console test SSL handshake with:
_https.get('https://<yourserver>/default.pdf', res => console.log(res.statusCode))_ 

If you find this error:
_Error: write EPROTO 140113041508160:error:14077410:SSL routines:SSL23_GET_SERVER_HELLO:sslv3 alert handshake failure:../deps/openssl/openssl/ssl/s23_clnt.c:802:_ and using nodejs 8 

Solution which worked for me after 3h fight was to add to this file:
_/usr/share/meteor/bundle/main.js_

This line on top:
require("tls").DEFAULT_ECDH_CURVE = "auto"

then restart BBB

Related bug:
https://github.com/nodejs/node/issues/16196

**Comment by teenkevo** [comment]
> # As of September 30, 2021
> this may be related to the expiration of the [DST Root CA X3 certificate](https://letsencrypt.org/docs/dst-root-ca-x3-expiration-september-2021).
> 
> ### To check if this is the case
> run `journalctl -f -u bbb-html5` and restart BigBlueButton: `bbb-config --restart` If you see the lines like
> 
> > systemd_start.sh[30265]: error: No file found. Error: certificate has expired
> > systemd_start.sh[30265]: error: Error parsing image size. Error: certificate has expired.
> 
> then it may be the case.
> 
> ### To fix the problem
> 1. Make sure the ISRG Root X1 certificate is present on your system:
>    `grep ISRG_Root_X1 /etc/ca-certificates.conf`
>    If it is not there, you should upgrade the ca-certificates package and the OpenSSL to the latest version possible.
> 2. Comment out the expired DST Root CA X3 certificate in `/etc/ca-certificates.conf` with an exclamation mark like this:
>    `!mozilla/DST_Root_CA_X3.crt`
> 3. Update the ca-certificates:
>    `sudo update-ca-certificates`
> 4. Make the Node.js use the system-wide CA certificates. For this purpose at the end of the file `/usr/share/meteor/bundle/systemd_start.sh` add the parameter `--use-openssl-ca` like this:
>    `PORT=3000 /usr/share/$NODE_VERSION/bin/node --use-openssl-ca --max-old-space-size=4096 --max_semi_space_size=128  main.js`
> 
> After that restart the BigBlueButton again, start some session and watch the logs. The lines about the expired certificate should not appear. At least, it worked in my case. Hope that helps somebody.

You need a git trophy

**Comment by khoubbakht** [comment]
> Hello Capilkey,
> 
> > I don't think "no_proxy" in npm would have any relevance here. It seems like that setting is for downloading node packages through a proxy, whereas the problem you're hitting is when the running application tries to download the presentation SVGs.
> > I still think the problem is going to be something about misconfigured or malformed certificates.
> 
> You are correct, it is a certificate configuration problem in nginx.
> 
> **Problem solved!**
> 
> When tracing the error, I came across the "https://www.npmjs.com/package/ssl-root-cas#unable-to-verify-the-first-certificate" link, where it explicitly says the following: "did do not include the intermediate certificates in the certificate it sent you." for this error.
> 
> I created the .pem file with the public key of the site + intermediate certificate. I replaced the .crt file with the public key of the site certificate with the two public keys in a .pem file.
> 
> The interesting thing is that:
> 
> A) For the use of the BBB with plugin in the browser, just the public key of the site certificate in nginx for the perfect operation.
> 
> B) **For html5 and Node,** public key information for the site and intermediate CA is required in a single .pem file.
> 
> The documentation says about the intermediate certificate, but I do not understand ...
> 
> #ssl_certificate /etc/nginx/ssl/nw-pbh.gov.br.crt; ssl_certificate /etc/nginx/ssl/com_intermediate_pbh.gov.br.pem;
> 
> I exported the keys from the browser, as pictured.

@lgmatabh 
Thank you!

**Comment by khoubbakht** [comment]
@lgmatabh 
Thank you!
